contrib/src/main/java/macrobase/analysis/transform/aggregate/BatchWindowAvg.java

```diff
@@ -0,0 +1,39 @@
+package macrobase.analysis.transform.aggregate;
+
+import macrobase.conf.ConfigurationException;
+import macrobase.conf.MacroBaseConf;
+import macrobase.conf.MacroBaseDefaults;
+import macrobase.datamodel.Datum;
+import org.apache.commons.math3.linear.ArrayRealVector;
+import org.apache.commons.math3.linear.RealVector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BatchWindowAvg extends BatchWindowAggregate {
+    public BatchWindowAvg(){}
+
+    public BatchWindowAvg(MacroBaseConf conf) throws ConfigurationException {
+        this.timeColumn = conf.getInt(MacroBaseConf.TIME_COLUMN, MacroBaseDefaults.TIME_COLUMN);
+    }
+
+    public Datum aggregate(List<Datum> data) {
+        if (data.isEmpty())
+            return new Datum(new ArrayList<>(), new ArrayRealVector(dim));
+
+        dim = data.get(0).metrics().getDimension();
+        RealVector results = new ArrayRealVector(dim);
```

Comment:
Doesn't seem like we benefit much from using a RealVector here over a double[], double[] is going to be faster (can just create an ArrayRealVector from it in the end)